dotnet3.5/S3/DeleteBucket/DeleteBucket.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX - License - Identifier: Apache - 2.0
+// snippet-start:[S3.dotnet35.DeleteBucket]
+using Amazon;
+using Amazon.S3;
+using System;
+using System.Threading.Tasks;
+
+namespace DeleteBucket
+{
+    class DeleteBucket
+    {
+        // Change the name of the following constant to the region containing the bucket
+        // The value below is just an example.
+        private static readonly RegionEndpoint BUCKET_REGION = RegionEndpoint.USWest2;
+        private static IAmazonS3 _s3Client;
+
+        // Specify the name of the bucket to delete
+        private const string BUCKET_NAME = "*** provide the name of the bucket with source object ***";
+
+        static void Main()
+        {
+            _s3Client = new AmazonS3Client(BUCKET_REGION);
+
+            // Now delete the bucket. If the bucket you are trying to
+            // delete contains any objects, the call will raise an exception.
```

Comment:
Perhaps provide an option to forcefully delete the bucket if there are objects in it? Of course, you'd also then need to add logic to check whether there indeed any objects in it, and then delete all of those objects.